server/src/core/user-notification/types.ts

```diff
@@ -0,0 +1,8 @@
+import { EventBus } from '../event';
+import { AtSmsProvider } from '../sms';
+
+export interface UserNotificationsArgs {
+  smsProvider: AtSmsProvider;
```

Comment:
This `smsProvider`'s type should be `SmsProvider` (the interface), not `AtSmsProvider` (the concrete AT-specific implementation)